### DISCUSSION TITLE
Early stopping can be added to avoid long training time

### DISCUSSION THREAD

**Discussion by AslantheAslan** [timeseriesAI/tsai#564] [Ideas]
I'm trying to get the best accuracy score out of all scopes in 100 epochs, for example. But I couldn't find out how to get the best accuracy. In other words, I wonder if there is an early stopping option in tsai API.

**Comment by oguiza** [comment]
Hi @AslantheAslan,
You can use fastai's EarlyStoppingCallback. 
Here's an example of how it could be used:
```python
from tsai.basics import *
from fastai.callback.tracker import EarlyStoppingCallback
X, y, splits = get_UCR_data('LSST', split_data=False)
tfms = [None, TSClassification()]
batch_tfms = TSStandardize(by_sample=True)
dls = get_ts_dls(X, y, splits=splits, tfms=tfms, batch_tfms=batch_tfms)
learn = ts_learner(dls, metrics=accuracy, cbs=[ShowGraph(), EarlyStoppingCallback(monitor='accuracy')])
learn.fit_one_cycle(2, 1e-2)
```

You can read the docs [here](https://docs.fast.ai/callback.tracker.html#earlystoppingcallback).

**Reply by AslantheAslan** [reply]
Thanks a lot for your time. Sorry if my comment was a bit unnecessary because I noticed that tsai models can be inferred by fastai API functions. I have implemented early stopping to my tsai model 1-2 days after adding the comment above. Thanks a lot for this wonderful library, again.
